src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -938,10 +958,21 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
 
     void TermControl::_KeyHandler(Input::KeyRoutedEventArgs const& e, const bool keyDown)
     {
+        if (e.OriginalKey() == VirtualKey::Enter)
+        {
+            return;
+        }
+
         // If the current focused element is a child element of searchbox,
         // we do not send this event up to terminal
-        if (_searchBox && _searchBox->ContainsFocus())
+        if (_searchBox && _searchBox->ContainsFocus() && _searchBox->TextBox().Text().empty())
+        {
+            return;
+        }
+
+        if (keyBindingSearch)
         {
+            keyBindingSearch = false;
             return;
         }

```

Comment:
This feels wrong to me. I know currently, actions are only triggerable with the keyboard, but theoretically, they might be invokable through other UI elements. Are we just trying to prevent the keys from being duplicated here?